File: backend/app/schemas/cost.py
"""Cost summary API schemas."""
from pydantic import BaseModel


class CostProviderSummary(BaseModel):
    """Monthly spend grouped by provider."""

    provider: str
    month_sek: float


class CostModelSummary(BaseModel):
    """Monthly spend grouped by model."""

    model_id: int
    model_name: str
    month_sek: float


class CostSummary(BaseModel):
    """Read-only cost dashboard response."""

    currency: str
    ceiling_sek: float
    month_spent_sek: float
    month_remaining_sek: float
    all_time_spent_sek: float
    by_provider: list[CostProviderSummary]
    by_model: list[CostModelSummary]

File: frontend/src/types/cost.ts
export type CostBreakdownByProvider = {
  provider: string
  month_sek: number
}

export type CostBreakdownByModel = {
  model_id: number
  model_name: string
  month_sek: number
}

export type CostSummary = {
  currency: 'SEK'
  ceiling_sek: number
  month_spent_sek: number
  month_remaining_sek: number
  all_time_spent_sek: number
  by_provider: CostBreakdownByProvider[]
  by_model: CostBreakdownByModel[]
}

export type CostCeilingResponse = {
  error: 'cost_ceiling_exceeded'
  spent_sek: number
  ceiling_sek: number
  currency: 'SEK'
}
